src/PostgREST/Error.hs

```diff
@@ -54,74 +54,107 @@ class (JSON.ToJSON a) => PgrstError a where
 
 
 
+
 data ApiRequestError
-  = ActionInappropriate
-  | InvalidRange
+  = ParseRequestError Text Text
+  | ActionInappropriate
   | InvalidBody ByteString
-  | ParseRequestError Text Text
-  | NoRelBetween Text Text Text
-  | AmbiguousRelBetween Text Text [Relationship]
-  | AmbiguousRpc [ProcDescription]
-  | NoRpc Text Text [Text] Bool ContentType Bool
+  | InvalidRange
+  | UnsupportedVerb -- Unreachable?
```

Comment:
I think this error is actually unreachable as mentioned in the comment. If the verb is not supported it would throw a `404 Not Found` before reaching the [`mutateRequest`](https://github.com/PostgREST/postgrest/blob/c858d158aa587510d8cf57373319f962514888ef/src/PostgREST/Request/DbRequestBuilder.hs#L312-L347) function. Doing this to the function, although less transparent, would omit using this error and give the same result (tests pass):

```haskell
-- mutateRequest function

    -- Change this:
    ActionDelete -> Delete qi <$> combinedLogic <*> pure returnings
    _            -> Left UnsupportedVerb

    -- Into this:
    _ -> Delete qi <$> combinedLogic <*> pure returnings
```

Reply:
I agree with the observation that we don't need this error. The suggested change, however, I don't agree with. This is really making it much less transparent. The proper solution would be to create an `ActionWrite WriteMethod` data constructor very similar to `ActionInvoke InvokeMethod`. This can the be passed from `writeQuery` in `App.hs` to `mutateRequest` - removing the impossible case.